stratum/lib/signal_waiter.cc

```diff
@@ -0,0 +1,71 @@
+// Copyright 2020-present Open Networking Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+#include "stratum/lib/signal_waiter.h"
+
+#include "stratum/glue/logging.h"
+#include "stratum/glue/status/status_macros.h"
+#include "stratum/public/lib/error.h"
+
+namespace stratum {
+namespace hal {
+
+namespace {
+
+// Signal received callback which is registered as the handler for signals using
+// signal() system call.
+void SignalRcvCallback(int value) { SignalWaiter::HandleSignal(value); }
+
+}  // namespace
+
+// Initialize static member.
+SignalWaiter SignalWaiter::instance_;
+
+SignalWaiter::SignalWaiter(const std::vector<int> signals) : signals_(signals) {
```

Comment:
Consider set to prevent duplicate signals and weird behavior after deconstruction.